src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -12906,5 +12951,15 @@ private new struct ResetPoint
             node = this.AddTrailingSkippedSyntax(node, trailingTrash.Node);
             return node;
         }
+
+        private SyntaxToken MergeTokens(SyntaxToken s1, SyntaxToken s2, SyntaxKind kind)
+        {
+            if (s1.GetTrailingTriviaWidth() == 0 && s2.GetLeadingTriviaWidth() == 0)
+            {
+                s1 = SyntaxFactory.Token(s1.GetLeadingTrivia(), kind, s2.GetTrailingTrivia());
+            }
```

Comment:
There should likely be an error or something here, or it should return a bool if it wasn't able to merge.